### ISSUE TITLE
Setting disappeared

### ISSUE THREAD

**Issue by DavidShusterman** [umbrellaplug/umbrellaplug.github.io#148]
Hey, I can't seem to use the settings. It just opens a blank popup. When clicking on the advanced it does open up the settings options but it won't let me authorize with real debrid. Clicking on the authorization button does nothing...

**Comment by umbrellaplug** [comment]
I've been fighting this for about a week. Can you get me your logs?

**Comment by DavidShusterman** [comment]
Log file is empty :/

**Comment by umbrellaplug** [comment]
Anything in the kodi.log?

**Comment by umbrellaplug** [comment]
Sorry to keep hitting on this one, but any information you can provide would be helpful. Type of device you're using, version of plugin, anything you think might help. I have been unable to duplicate it here to track it down.

**Comment by DavidShusterman** [comment]
I see some failures
https://paste.kodi.tv/fixinopifa

Let me know what else I can do
I'm with umbrella latest

Kodi 19.4
Compiled 15.10.22
Using LibreElec 10.0.3 kernel Linux 5.10.110

**Comment by umbrellaplug** [comment]
Thanks for the logs. They were helpful. It appears there's an issue accessing the settings file in the userdata folder. I've done some research on this and it appears it could be from some code I have to check settings accessing the settings too often on arm processors. I will be making some adjustments to how often I access settings to see if I can help. This is a core Kodi function so unfortunately there's not really a "fix" on my side for it other than accessing settings less often.

**Comment by umbrellaplug** [comment]
I'm pretty sure since it's been a day you've likely blown this out and re-installed (understandable), but if you have not and you're still in this state. If you could get your settings.xml from your userdata folder, I can check for corruption in that xml file that could potentially help me to identify which setting is being written/ read too often. Feel free to remove any strings you want from that file before sending if you do still have it. It will have some tokens and such for real-debrid and other services that are exclusive to you that you may want to remove if you care. 

I noticed your github, I'm hoping you might have a better understanding of the struggle of tracking these random issues down. 

In summary: Think the issue is a kodi bug of corrupting xml from reading and writing settings too much. We send and receive a settings object. We do not directly edit those files it's done by core kodi. I'm looking to reduce the amount of reads and writes to settings from addon by identifying the setting that's being corrupted. I suspect it's "play.mode.tv" setting.

**Comment by DavidShusterman** [comment]
Sure thing man, I'm here to help.
Sorry for the ignorance, how to I get the settings.xml file?

**Comment by umbrellaplug** [comment]
No worries. On libreelec the easiest way would be to smb into the device. 
In windows: \\IPORDEVICENAME\Userdata\addon_data\plugin.video.umbrella

There should be a settings.xml file there.

**Comment by umbrellaplug** [comment]
Did you have the Playnext feature in the addon enabled before this happened?

**Comment by DavidShusterman** [comment]
Ok, sorry for the delay.
The settings.xml file is empty for some reason. and as for the Playnext feature, no idea, not something I've turned on atleast.

**Comment by DavidShusterman** [comment]
<img width="855" alt="image" src="https://user-images.githubusercontent.com/16805596/208315605-fc795264-17ad-40c0-9e6a-7830036475e5.png">

**Comment by umbrellaplug** [comment]
Delete that file. That's the issue. Once you restart kodi the plugin should rebuild a new settings.xml and you should be fixed. I believe I have found what is causing this to happen as well. I'm going to push a patch in just a moment. I'll send you a link here.

**Comment by umbrellaplug** [comment]
Would you be interested in testing a fix?
